src/libraries/System.Private.CoreLib/src/System/Threading/LazyInitializer.cs

```diff
@@ -100,8 +104,12 @@ public static class LazyInitializer
         /// if an object was not used and to then dispose of the object appropriately.
         /// </para>
         /// </remarks>
-        public static T EnsureInitialized<T>([NotNull] ref T? target, Func<T> valueFactory) where T : class =>
-            Volatile.Read(ref target) ?? EnsureInitializedCore(ref target, valueFactory);
+        public static T EnsureInitialized<T>([NotNull] ref T? target, Func<T> valueFactory) where T : class
+        {
+            T value = Volatile.Read(ref target) ?? EnsureInitializedCore(ref target, valueFactory);
+            NullabilityHelpers.SuppressNonNullAssignmentWarning(ref target);
```

Comment:
Ditto.  If this is warning, that seems like a Roslyn bug, unless I'm missing something.

Reply:
(And assuming it is a Roslyn bug, I'd rather just use `!` to suppress it and add a `// TODO-NULLABLE: https://...` comment linking to the appropriate issue.)